config/pipelines/pipelines/meteor_jupyterhub.yaml

```diff
@@ -0,0 +1,151 @@
+        - name: subdirectory
+          value: repo
+
+    - name: generate
+      taskRef:
+        name: generate-jupyterhub
+      runAfter:
+        - git-clone
+      workspaces:
+        - name: data
+          workspace: data
+      params:
+        - name: url
+          value: $(params.url)
+
+    - name: extract-jupyterhub-namespace
+      taskRef:
+        name: extract-external-service
+      params:
+        - name: externalServices
+          value: "$(params.externalServices)"
+        - name: serviceName
+          value: jupyterhub
+        - name: serviceProperty
+          value: namespace
+
+    - name: buildah
+      taskRef:
+        name: buildah
+      runAfter:
+        - generate
+        - extract-jupyterhub-namespace
+      workspaces:
+        - name: data
+          workspace: data
+        - name: sslcertdir
+          workspace: sslcertdir
+      params:
+        - name: IMAGE
+          value: image-registry.openshift-image-registry.svc:5000/$(tasks.extract-jupyterhub-namespace.results.value)/$(context.pipelineRun.name)
+        - name: BASE_IMAGE
+          value: $(tasks.generate.results.baseImage)
+
+    - name: resolve-owner-reference
+      taskRef:
+        name: resolve-owner-reference
+      runAfter:
+        - extract-jupyterhub-namespace
+      params:
+        - name: ownerReferences
+          value: $(params.ownerReferences)
+        - name: namespace
+          value: $(tasks.extract-jupyterhub-namespace.results.value)
+
+    - name: create-image-stream
+      taskRef:
+        name: openshift-client
+        kind: ClusterTask
+      runAfter:
+        - buildah
+        - resolve-owner-reference
+      params:
+        - name: SCRIPT
+          value: |
+            cat <<EOM | oc apply -f -
+            ---
+            kind: ImageStream
+            apiVersion: image.openshift.io/v1
+            metadata:
+              annotations:
+                opendatahub.io/notebook-image-name: $(context.pipelineRun.name)
+                opendatahub.io/notebook-image-url: $(params.url)
+              name: $(context.pipelineRun.name)
+              namespace: $(tasks.extract-jupyterhub-namespace.results.value)
+              labels:
+                opendatahub.io/notebook-image: 'true'
+            spec:
+              lookupPolicy:
+                local: true
+              tags:
+                - name: latest
+            EOM
+
+            oc patch -n $(tasks.extract-jupyterhub-namespace.results.value) imagestream/$(context.pipelineRun.name) --type='merge' -p='$(tasks.resolve-owner-reference.results.patch)'
+
+    - name: url-to-results
+      taskRef:
+        name: oc-actions
+      runAfter:
+        - buildah
+      params:
+        - name: script
+          value: |
+            echo "http://$(oc get route jupyterhub -n $(tasks.extract-jupyterhub-namespace.results.value) --template='{{"{{"}} {{"."}}spec.host {{"}}"}}')/hub/spawn#:~:text=$(context.pipelineRun.name)" > $(results.results.path)
+
```

Comment:
I believe this is not necessary in the context of opendatahub, as the spawn URL does not have to be shown in a (non-existent) Meteor UI